stdlib/SparseArrays/docs/src/AbstractSparseMatrixCSC.md

```diff
@@ -0,0 +1,59 @@
+    AbstractSparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
+
+Supertype for matrix with compressed sparse column (CSC).
+
+## `AbstractSparseMatrixCSC` interface
+
+In addition to the [Abstract array interface](@ref man-interface-array), every
+`AbstractSparseMatrixCSC` subtype `TS` must provide the following methods:
+
+* [`size(::TS)`](@ref size)
+* [`getcolptr(::TS) :: AbstractVector{<:Ti}`](@ref getcolptr)
+* [`rowvals(::TS) :: AbstractVector{<:Ti}`](@ref rowvals)
+* [`nonzeros(::TS) :: AbstractVector{<:Tv}`](@ref nonzeros)
+
+Other sparse matrix methods such as [`nzrange`](@ref) and [`nnz`](@ref) are automatically
+defined in terms of above functions.
+
+## Assumed invariance
+
+To use algorithms defined in SparseArrays, a matrix `A` of type `AbstractSparseMatrixCSC`
+must satisfy the following constraints.
+
+### Matrix `A` and all vectors constituting `A` have one-based indexing
```

Comment:
I don't think this is not necessary in principle.  But I think most of the code in SparseArrays.jl requires it so I think it's better to specify it.